javaagent-extensions/gcp-auth/src/test/java/com/google/cloud/opentelemetry/extension/auth/ExtensionIntegrationTest.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.opentelemetry.extension.auth;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.google.cloud.opentelemetry.extension.auth.testbackend.DummyOTelHttpEndpoint;
+import com.sun.net.httpserver.HttpServer;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.ProcessBuilder.Redirect;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// TODO: Fix testing harness
+public class ExtensionIntegrationTest {
+
+  private static final String TEST_SERVER_PATH = "/doWork";
+
+  private static final Logger logger = LoggerFactory.getLogger(ExtensionIntegrationTest.class);
+
+  @Test
+  public void smokeTest() throws IOException, InterruptedException {
+    String testAppJarPath =
+        new File("./build/libs/auto-instrumented-test-server.jar").getAbsolutePath();
+    // String dummyBackendJarPath = new
+    // File("./build/libs/dummy-otlp-backend.jar").getAbsolutePath();
+    String javaAgentJarPath = new File("./build/libs/otel-agent.jar").getAbsolutePath();
+    String authExtensionJarPath = new File("./build/libs/gcp-auth-extension.jar").getAbsolutePath();
+
+    DummyOTelHttpEndpoint otlp = new DummyOTelHttpEndpoint();
+    HttpServer backendServer;
+    try {
+      backendServer = otlp.createServer();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    backendServer.start();
+    Process p =
```

Comment:
Did you ever get the tests working?

Reply:
Got it working now - for some unknown reason, I could not get the @WithSpan annotation to work, so I switched to a different library which I know is auto-instrumented.